react-backend/client/src/components/admin/adminUsersPage.js

```diff
@@ -85,23 +83,43 @@ export default class AdminUsersPage extends Component {
   getActivitiesForChart = (activities) => {
     if (!activities) return [];
 
-    return activities.map(({ name, count }, index) => ({
-      value: count,
-      color: this.getColorPair(index).color,
-      highlight: this.getColorPair(index).highlight,
-      label: name,
-    }));
+    const activitiesData = {
+      labels: [],
+      datasets: [
+        {
+          data: [],
+          label: [],
+        },
+      ],
+    };
+
+    activities.forEach(({ name, count }) => {
+      activitiesData.datasets[0].data.push(count);
```

Comment:
Is this the correct shape of the data? `activitiesData` is passed in on line 37:

https://github.com/TwinePlatform/DataPower/blob/5bc5e8b413a328d9740a56644f4fedb3f21956e9/react-backend/client/src/components/admin/adminUsersPage.js#L37

The API documentation says that `activitiesNumbers` on this line looks like `[{ name, count},...]`. There's nothing about a dataset field, where's that coming from?

Reply:
- By the Adapter Pattern do you mean pulling out the code to a separate class/file?

- I think I'm going to remove colours for now anyway as its something we'll need to properly think about later.

-The chart api needs all data to be within a datasets field to be able to work properly especially as we're using labels as the legend.

Reply:
> The chart api needs all data to be within a datasets field to be able to work properly especially as we're using labels as the legend.

Sorry, my mistake, I skipped over the lines where you initialised your datastructure. Nevermind.

> By the Adapter Pattern do you mean pulling out the code to a separate class/file?

I had thought about pulling all the conversion code out into the API client, but have since changed my mind, since it's far to heavily coupled to chart.js, which is specific to a few components.

Basically, ignore my previous comments.